app/services/bulk_segmented_users.rb

```diff
@@ -0,0 +1,43 @@
+class BulkSegmentedUsers
```

Comment:
TODO: Need to add docs, but this class was necessary because by default Rails will insert the join table records one by one which is not ideal to say the least if we're going to be processing up to 10k users at a time